HW1.py

```diff
@@ -3,90 +3,75 @@
 import pandas as pd
 import plotly.express as px
 from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import accuracy_score
 from sklearn.neighbors import KNeighborsClassifier
 from sklearn.pipeline import Pipeline
 from sklearn.preprocessing import StandardScaler
 from sklearn.tree import DecisionTreeClassifier
 
 
-def main():
-
+def load_data(data_file):
     # Load Data
-    iris_data = pd.read_csv("iris.data")
-
-    # Rename columns
-    iris_data = iris_data.rename(
-        columns={
-            "5.1": "sepal_length",
-            "3.5": "sepal_width",
-            "1.4": "petal_length",
-            "0.2": "petal_width",
-        }
-    )
+    iris_data = pd.read_csv("iris.data", header=None)
+    # name columns
+    iris_data.columns = [
+        "sepal_length",
+        "sepal_width",
+        "petal_length",
+        "petal_width",
+        "class",
+    ]
+    return iris_data
 
-    # Summary statistics
-    summary = pd.DataFrame.describe(iris_data)
-    print(summary)
 
-    # Plot and display figures
-    fig_hist = px.histogram(iris_data, x="sepal_length", color="Iris-setosa")
+def summary(df):
+    return df.describe()
+
+
+def plot(df):
+    fig_hist = px.histogram(df, x="sepal_length", color="class", opacity=0.5)
```

Comment:
`opacity=` appears to control transparency for the entire graph, not individual classes.

Based on [this documentation](https://plotly.com/python/histograms/#overlaid-histogram), you may want to switch this `barmode="overlay"`

Reply:
Great suggestion! Fixed.

Reference: 74608ada84ac5b1ba2c16a8efccceaab33ff438b